core/common/src/main/java/alluxio/collections/IndexedSet.java

```diff
@@ -375,7 +503,21 @@ public int size() {
    * @param value the field value
    * @return the set of objects with the specified field value
    */
-  private ConcurrentHashSet<T> getByFieldInternal(FieldIndex<T> index, Object value) {
-    return mIndexMap.get(index).get(value);
+  private T getByFieldInternalUnique(FieldIndex<T> index, Object value) {
```

Comment:
why not just make the parameter `UniqueFieldIndex` instead of `FieldIndex`? In that case, you can ensure type-safety without the following `checkState`